src/Table.elm

```diff
@@ -421,6 +423,22 @@ view (Config { toId, toMsg, columns, customizations }) state data =
     sortedData =
       sort state columns data
 
+    caption =
+      case customizations.caption of
+        Nothing ->
+          identity
+
+        Just { attributes, children } ->
+          (::) <| Html.caption attributes children
```

Comment:
I think it's harder to read your way. What I have takes more lines, but is easier to follow.